python/dgl/graphbolt/itemset.py

```diff
@@ -85,9 +85,16 @@ def __iter__(self) -> Iterator:
         if len(self._items) == 1:
             yield from self._items[0]
             return
-        zip_items = zip(*self._items)
-        for item in zip_items:
-            yield tuple(item)
+
+        if isinstance(self._items[0], Sized):
+            items_len = len(self._items[0])
+            for i in range(items_len):
```

Comment:
add comment here to explain why we iterate with explicit loop instead of `zip` here: performance, especially the first batch access. add details here alongside the code makes the code more readable.

Reply:
even could link to a ticket which records more details including the perf comparison log.

Reply:
> even could link to a ticket which records more details including the perf comparison log.

I think I can link to the PR rather than my gist. I think this description is detailed enough.